services/workflows-service/plugins/check-repository-args/check-repository-args.rule.js

```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  meta: {
+    type: 'problem',
+    severiry: 'error',
+    docs: {
+      description: 'Check if repository methods receive projectId: TProjectId',
+      category: 'Best Practices',
+      recommended: true,
+    },
+  },
+  create: function (context) {
+    const isRepository = /^(?!.*\/customer\.repository\.(ts|js)$).*\.repository\.(ts|js)$/.test(
+      // Using deprecated .getFilename here because relevant context.filename returns undefined
+      context.getFilename(),
+    );
+
+    return {
+      MethodDefinition: function (node) {
+        if (!isRepository || node.key.name === 'constructor') return;
+        const isUnscoped = node.key.name.toLowerCase().includes('unscoped');
+
+        if (isUnscoped) return;
+
+        const isProjectIdsIncluded =
+          node.value.params.length &&
```

Comment:
Can be simplified, length check is needed for `every` and not `some`.

Reply:
Good point,thanks